src/services/PwmService/PwmService.ts

```diff
@@ -0,0 +1,191 @@
+  }
+
+  /**
+   * @async Calls the PWM get_duty_cycle SDK method through an RPC request
+   */
+  async getDutyCycle(PWM: PWM): Promise<number> {
+    const response = await this.PWMRequest("get_duty_cycle", PWM, GET_DUTY_CYCLE);
+    return (response.content as GetDutyCycle).dutyCycle;
+  }
+
+  /**
+   * @async Calls the PWM get_polarity SDK method through an RPC request
+   */
+  async getPolarity(PWM: PWM): Promise<Polarity> {
+    const response = await this.PWMRequest("get_polarity", PWM, GET_POLARITY);
+    return (response.content as GetPolarity).polarity;
+  }
+
+  /**
+   * @async Calls the PWM get_enabled SDK method through an RPC request
+   */
+  async getEnabled(PWM: PWM): Promise<boolean> {
+    const response = await this.PWMRequest("get_enabled", PWM, GET_ENABLED);
+    return (response.content as GetEnabled).enabled;
+  }
+
+  /**
+   * @async Calls the PWM set_config SDK method through RPC
+   */
+  async setConfig({
+    pwmNum,
+    frequency = undefined,
+    dutyCycle = undefined,
+    polarity = undefined,
+  }: PWMConfig): Promise<string> {
+    const requestType = this.serviceProtoRoot.lookupType(
+      CONFIG_TYPE
+    );
+    const responseType = this.serviceProtoRoot.lookupType(
+      SUCCESS_MSG
+    );
+    const argsList = createConfigArgsList({ pwmNum, frequency, dutyCycle, polarity });
+
+    const response = await this.handlePWMRequest(
+      requestType,
+      responseType,
+      "set_config",
+      { confArg: argsList }
+    );
+    return (response.content as SuccessMsg).content;
+  }
+
+  /**
+   * @async Sends an RPC request for a specified PWM SDK method
+   */
+  private async PWMRequest(
+    methodName: string,
+    pwmNum: PWM,
+    responseTypeName: string,
+  ): Promise<serverResponse> {
+    const requestType = this.serviceProtoRoot.lookupType(
+      PWM_TYPE
+    );
+    const responseType = this.serviceProtoRoot.lookupType(
+      responseTypeName
+    );
+
+    return await this.handlePWMRequest(
+      requestType,
+      responseType,
+      methodName,
+      { pwmNum }
+    );
+  }
+
+  /**
+   * @async Handles an RPC request and returns a response message
+   */
+  private async handlePWMRequest(
+    requestType: protobuf.Type,
+    responseType: protobuf.Type,
+    methodName: string,
+    requestMsg: Object
+  ): Promise<serverResponse> {
+    const serviceReq: serviceRequest = {
+      serviceName: this.serviceName,
+      methodName,
+      requestMsg,
+    };
+
+    console.info(
+      `Calling PWM ${methodName.replace(/_([a-z])/g, (_, group1) =>
+        group1.toUpperCase()
+      )}` + ` through Rpc Channel.`
+    );
+
+    const response: serverResponse = await this.rpcChannel.callMethod(
+      serviceReq,
+      requestType,
+      responseType
+    );
+
+    if (!response || response.error !== undefined) {
+      throw new Error(response ? response.error : "Response is null or undefined.");
+    }
+    return response;
```

Comment:
These lines of code are a bit clunky (the condition in the IF block covers the condition of the exception). I think you can improve it. For example:

  if (response && !response.error) { //Return response if there is a response and NO error
      return response;
   } else {
    // Return any available error OR if the response is not available
      throw new Error(response && response.error || "Response is null or undefined.");
    }

Reply:
Thanks, that makes sense.